velox/functions/prestosql/window/tests/FirstLastValueTest.cpp

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/prestosql/window/tests/WindowTestBase.h"
+#include "velox/vector/fuzzer/VectorFuzzer.h"
+
+using namespace facebook::velox::exec::test;
+
+namespace facebook::velox::window::test {
+
+namespace {
+
+static const std::vector<std::string> kValueFunctions = {
+    std::string("first_value(c1)"),
+    std::string("first_value(c2)"),
+    std::string("last_value(c1)"),
+    std::string("last_value(c2)")};
+
+struct TestParam {
+  const std::string valueFunction;
+  const std::string frameClause;
+};
+
+std::vector<TestParam> getTestParams(
+    const std::vector<std::string>& frameClauses) {
+  std::vector<TestParam> params;
+  for (auto valueFunction : kValueFunctions) {
+    for (auto frameClause : frameClauses) {
+      params.push_back({valueFunction, frameClause});
+    }
+  }
+  return params;
+}
+
+class FirstLastValueTest : public WindowTestBase {
```

Comment:
Can this test be consolidated with the NthValue test ? Would be great to have all the functions in one file.

Reply:
Thanks for the suggestion. Merged the tests with NthValueTest and renamed file to ValueFunctionTest.cpp, will rebase PR after the WindowTestBase refactoring changes are merged.